extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/KafkaRuntimeConfigProducer.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.kafka.client.runtime;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.StreamSupport;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Produces;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.arc.DefaultBean;
+
+@Dependent
+public class KafkaRuntimeConfigProducer {
+
+    private String configPrefix = "kafka";
+
+    @Produces
+    @DefaultBean
+    @ApplicationScoped
+    public KafkaRuntimeConfig createKafkaRuntimeConfig() {
+        Map<String, String> properties = new HashMap<>();
+        final Config config = ConfigProvider.getConfig();
+
+        StreamSupport
+                .stream(config.getPropertyNames().spliterator(), false)
+                .map(String::toLowerCase)
+                .filter(name -> name.startsWith(configPrefix))
```

Comment:
It can be confusing as the Kafka Client is not produced by default. Typically, it would not cover the configuration coming from reactive messaging. Existing applications create their properties directly (following the Kafka documentation).

Reply:
Okay, in that case I could remove the KafkaRuntimeConfiguration again and would only have to make sure all the clients in integration-tests/kafa get the right set of properties. I just found it to be very cumbersome to inject them one by one, especially when extending the test case to more than plain authentication.